src/libsystemd/sd-journal/sd-journal.c

```diff
@@ -2236,26 +2236,24 @@ _public_ int sd_journal_get_monotonic_usec(sd_journal *j, uint64_t *ret, sd_id12
         return 0;
 }
 
-static bool field_is_valid(const char *field) {
-        const char *p;
-
+static bool field_is_valid(const char *field, size_t field_length) {
         assert(field);
 
-        if (isempty(field))
+        if (field_length == 0)
                 return false;
 
```

Comment:
maybe add `if (field_length == SIZE_MAX) field_lenght = strlen(field)` here? And then just call the functions with SIZE_MAX as second parameter?